### PR TITLE
iOS Xcode 6.3 support

### PR DESCRIPTION
**Pull request by mviamari**
I've attempted to update the Example App for iOS to support XCode 6.3 and Swift 1.2

Mostly it seemed to be XCode 6.3 prefers Int to UInt, and a few other calls around countElements(Array) which generally become Array.count.

I did not attempt to update the Mac example App.

### CODE REVIEW

File: DominantColor/Shared/DominantColors.swift [hunk 1]
```diff
@@ -161,19 +161,19 @@ private func distanceForAccuracy(accuracy: GroupingAccuracy) -> (INVector3, INVe
     case .Low:
         return CIE76SquaredColorDifference
     case .Medium:
-        return CIE94SquaredColorDifference()
+        return { CIE94SquaredColorDifference()(colors: $0) }
```
Comment (indragiek) [inline DominantColor/Shared/DominantColors.swift]: Is this change necessary? I don't recall anything in Swift 1.2 that would make the previous implementation not compile.

Comment (mviamari) [inline DominantColor/Shared/DominantColors.swift]: I also thought the same thing. I don't recall any explicit changes that would affect this, but it is causing a compile error for me.

The basic issue is that the function signature for the three internal functions are different, and this creates the compiler error. CIE76SquaredColorDifference defines a signature with two ```INVector3``` as parameters, and the other two define a signature with a single tuple parameter made up of two ```INVector3``` objects.

```
DominantColor/DominantColor/Shared/DominantColors.swift:166:16: Function signature '(colors: (INVector3, INVector3)) -> Float' is not compatible with expected type '(INVector3, INVector3) -> Float'
```

The issue could be also be fixed by altering the function signature for ```CIE94SquaredColorDifference``` and ```CIE2000SquaredColorDifference``` to expect two distinct parameters rather than a single tuple.

I can easily alter this to change the function signatures rather than the calling pattern if desired.

Comment (indragiek) [inline DominantColor/Shared/DominantColors.swift]: Thank you for the explanation. You can change those function signatures then :+1:

**Comment by mviamari** [comment]
Updated.  All the color difference functions now expect two distinct parameters of type INVector3.

**Comment by indragiek** [comment]
Thank you, looks good!
